src/test/java/org/openmrs/addonindex/domain/AddOnInfoAndVersionsTest.java

```diff
@@ -16,29 +17,43 @@
 @RunWith(SpringRunner.class)
 @JsonTest
 public class AddOnInfoAndVersionsTest {
-	
-	@Autowired
-	private ObjectMapper mapper;
-	
-	@Autowired
-	private JacksonTester<AddOnInfoAndVersions> json;
-	
-	@Test
-	public void testSerialization() throws Exception {
-		AddOnVersion version = new AddOnVersion();
-		version.setVersion(new Version("1.0"));
-		
-		AddOnInfoAndVersions info = new AddOnInfoAndVersions();
-		info.setName("OpenMRS");
-		info.setDescription("Write code, save lives");
-		info.addVersion(version);
-		
-		String json = mapper.writeValueAsString(info);
-		AddOnInfoAndVersions parsed = mapper.readValue(json, AddOnInfoAndVersions.class);
-		assertThat(parsed.getName(), is("OpenMRS"));
-		assertThat(parsed.getDescription(), is("Write code, save lives"));
-		assertThat(parsed.getVersions().size(), is(1));
-		assertThat(parsed.getVersions().get(0).getVersion().toString(), is("1.0"));
-	}
-	
+
+    @Autowired
+    private ObjectMapper mapper;
+
+    @Autowired
+    private JacksonTester<AddOnInfoAndVersions> json;
+
+    @Test
+    public void testSerialization() throws Exception {
+        AddOnVersion version = new AddOnVersion();
+        version.setVersion(new Version("1.0"));
+
+        AddOnInfoAndVersions info = new AddOnInfoAndVersions();
+        info.setName("OpenMRS");
+        info.setDescription("Write code, save lives");
+        info.addVersion(version);
+
+        String json = mapper.writeValueAsString(info);
+        AddOnInfoAndVersions parsed = mapper.readValue(json, AddOnInfoAndVersions.class);
+        assertThat(parsed.getName(), is("OpenMRS"));
+        assertThat(parsed.getDescription(), is("Write code, save lives"));
+        assertThat(parsed.getVersions().size(), is(1));
+        assertThat(parsed.getVersions().get(0).getVersion().toString(), is("1.0"));
+    }
+
+    @Test
+    public void testGetLatestMatchedVersion() throws Exception {
+
+        AddOnVersion version = new AddOnVersion();
+        version.setVersion(new Version("1.0"));
+        version.setRequireOpenmrsVersion("2.1.0");
+
+        AddOnInfoAndVersions info = new AddOnInfoAndVersions();
+        info.setName("OpenMRS");
+        info.setDescription("Write code, save lives");
+        info.addVersion(version);
+        assertEquals("1.0", info.getLatestSupportedVersion("2.1.0").getVersion().toString());
```

Comment:
The correct assertion here would be to check that what's returned here is the same object as you've defined above, not just comparing the string for equality.

    assertEquals(version, info.getLatestSupportedVersion("2.1.0"));

or the more modern way is

    assertThat(info.getLatestSupportedVersion("2.1.0"), is(version));